crates/cheatcodes/src/inspector.rs

```diff
@@ -1165,11 +1176,15 @@ impl<DB: DatabaseExt> Inspector<DB> for Cheatcodes {
         self.gas_metering.last_call_gas = Some(crate::Vm::Gas {
             gasLimit: gas.limit(),
             gasTotalUsed: gas.spent(),
-            gasMemoryUsed: 0,
             gasRefunded: gas.refunded(),
             gasRemaining: gas.remaining(),
         });
 
+        // Store the total gas used for all active gas records started by `startSnapshotGas`.
+        self.gas_metering.gas_records.iter_mut().for_each(|record| {
```

Comment:
looks like this will summarize gas usage for calls of any depth, thus resulting in deeper calls usage being added twice?

Reply:
also I believe it wouldn't be possible to use `startSnapshotGas` for snapshotting internal libraries (because we only account for calls), are we OK with this limitation?

Reply:
That's correct and is one of my concerns, one can work around this limitation by using `gasleft()`'s with `snapshotValue`. Do you see a way we could add this?

Reply:
I think a cheap way of doing that would be to also add similar logic to `step_end` hook where we have interpreter keeping gas usage data of the current call

though might make sense to consider making this an opt-in

Reply:
I think with the current limitation of just tracking external calls it is strictly worse in terms out accuracy than the existing `forge-snapshot` which means this would be a blocker because users will except at least feature parity

Reply:
> shouldn't it yield just 53874?

ah.. as in `53874` contains `27491` and `22312` - I see now